contribs/gmf/src/floor/floorselectorComponent.js

```diff
@@ -0,0 +1,169 @@
+    return templateUrl !== undefined ? templateUrl :
+      'gmf/floor/floorselectorcomponent';
+  });
+
+
+/**
+ * @param {!angular.IAttributes} $attrs Attributes.
+ * @param {!function(!angular.IAttributes): string} gmfFloorselectorTemplateUrl Template function.
+ * @return {string} Template URL.
+ * @ngInject
+ * @private
+ * @hidden
+ */
+function gmfFloorselectorTemplateUrl($attrs, gmfFloorselectorTemplateUrl) {
+  return gmfFloorselectorTemplateUrl($attrs);
+}
+
+
+/**
+ * @private
+ * @hidden
+ */
+class Controller {
+
+  /**
+   * @param {!angular.IScope} $scope Angular scope.
+   * @param {!JQuery} $element Element.
+   * @private
+   * @ngInject
+   * @ngdoc controller
+   * @ngname GmfFilterselectorController
+   */
+  constructor($scope, $element) {
+
+    /**
+     * @type {Array.<Object.<string, string>>}
+     */
+    this.items;
+
+    /**
+     * @type {number}
+     */
+    this.size;
+
+    /**
+     * @type {string}
+     */
+    this.floor;
+
+    /**
+     * @type {number}
+     */
+    this.currentIndex;
+
+    /**
+     * @type {number}
+     * @private
+     */
+    this.lowerBound_;
+
+    /**
+     * @type {number}
+     * @private
+     */
+    this.upperBound_;
+
+    this.scope = $scope;
+    this.element = $element;
+  }
+
+  $postLink() {
+    this.scope.$watch(
+      () => {
+        return this.floor;
+      },
+      () => {
+        this.floorChanged_();
+      }
+    );
+
+    this.element[0].addEventListener('wheel', (event) => {
+      this.scope.$apply(() => {
+        const delta = event.deltaY > 0 ? -1 : 1;
+        this.move(delta);
+      });
+    });
+  }
+
+  /**
+   * @private
+   */
+  floorChanged_() {
+    const floor = this.floor;
+
+    // Update currentIndex
+    this.currentIndex = findIndexInArray(this.items, function(item) {
+      return item.value === floor;
+    });
+    console.assert(this.currentIndex > -1);
+
+    const buttonGroup = this.element.find('.btn-group-floors');
+
+    const buttonUp = this.element.find('.btn-floor-up');
+    const maxTop = buttonUp.position().top + buttonUp.outerHeight(true);
+
+    const buttonDown = this.element.find('.btn-floor-down');
+    const minTop = buttonDown.position().top - buttonGroup.outerHeight(true);
+
+    const currentButton = this.element.find(`.btn-floor:nth(${this.currentIndex})`);
+    let top = (
+      this.element.innerHeight() / 2
+      - currentButton.position().top
+      - currentButton.outerHeight(true) / 2
+    );
+    top = Math.min(top, maxTop);
+    top = Math.max(top, minTop);
+    buttonGroup.css('top', top);
+  }
+
+  /**
+   * @param {number} delta Signed number of floors to move.
+   */
+  move(delta) {
+    const newindex = this.currentIndex - delta; // Items are in reversed order
+    if (newindex >= 0 && newindex < this.items.length) {
+      this.floor = this.items[newindex].value;
+    }
+  }
+}
+
+
+module.component('gmfFloorselector', {
+  bindings: {
+    floor: '=',
+    items: '<',
+    size: '<'
```

Comment:
Where is this variable used?

Reply:
Good catch, it was the number of visible items in first version, now size is given in css.
I will remove it.